File: dotnet/SOLID/PaymentProcessor.cs
using dotnet.SOLID.Interface;
using dotnet.SOLID.Models;

namespace dotnet.SOLID;

public class PaymentProcessor
{
    private readonly IPaymentStrategy _paymentProcessor;
    
    public PaymentProcessor(IPaymentStrategy paymentProcessor)
    {
        _paymentProcessor = paymentProcessor;
    }
    
    public void ProcessPayment(Order order)
    {
        Console.WriteLine("processing payment for order " + order.Name + "with price " + order.Price + " Now!");
        
        _paymentProcessor.ProcessPayment(order.Price);
    }
}

File: typescript/solid/PaymentProcessor.ts
import { Order } from "./Order";
import { Payment } from "./Payment";
import { PaymentStrategy } from "./PaymentStrategy";

export class PaymentProcessor {

    private paymentStrategy: PaymentStrategy;

    constructor(paymentStrategy: PaymentStrategy) {
        this.paymentStrategy = paymentStrategy;
    }

    processPayment(order: Order): void {
        console.log("Processing payment of order: " + order.getName());
        console.log("Issuing payment for amount: " + order.getPrice());
        this.paymentStrategy.processPayment(order.getPrice());
    }

}